classes/class-gf-securesubmit.php

```diff
@@ -1668,4 +1674,570 @@ public function get_validation_result($validationResult, $authorizationResult)
+
+            $this->log_debug(__METHOD__ . '(): Create payment method.');
+            $paymentMethod = $this->createPaymentMethod($payPlanCustomer);
+            /** @var HpsPayPlanPaymentMethod $payPlanPaymentMethod */
+            $payPlanPaymentMethod = $payPlanService->addPaymentMethod($paymentMethod);
+
+            if (null === $payPlanPaymentMethod->paymentMethodKey) {
+
+                $this->log_debug(__METHOD__ . '(): Could not create Pay Plan payment method');
+
+                return $this->authorization_error($userError);
+
+            }
+
+            // Get HPS plan for feed.
+            $this->log_debug(__METHOD__ . '(): Create Schedule.');
+            /** @var HpsPayPlanSchedule $plan */
+            $plan = $this->create_plan($payPlanPaymentMethod, $feed, $payment_amount, $trial_period_days,
+                $currency);;
+
+            // If error was returned when retrieving plan, return plan.
+
+            /** @var HpsPayPlanSchedule $planSchedule */
+            $this->log_debug(__METHOD__ . '(): Add Schedule.');
+            $planSchedule = $payPlanService->addSchedule($plan);
+
+            // Create the plan unless there is no key.
+            if (null === $planSchedule->scheduleKey) {
+
+                $this->log_debug(__METHOD__ . '(): Could not create Pay Plan Schedule');
+
+                return $this->authorization_error($userError);
+
+            }
+
+            if (null !== $trial_period_days) {
+
+                $this->log_debug(__METHOD__ . '(): Processing one time setup fee');
+                /** @var HpsAuthorization $response */
+                /** @noinspection PhpParamsInspection */
+                $response = $this->processRecurring($payment_amount, $feed, $payPlanPaymentMethod,
+                    $planSchedule);
+
+                if (!($response->transactionId > 0 && null !== $response->authorizationCode)) {
+
+                    $this->log_debug(__METHOD__ . '(): First Charge Failed!! ');
+
+                    return $this->authorization_error($userError);
+
+                }
+
+            } // if
+
+            // If a setup fee is required, add an invoice item.
+            if ($single_payment_amount) {
+
+                $this->log_debug(__METHOD__ . '(): Processing one time setup fee');
+                /** @var HpsAuthorization $response */
+                /** @noinspection PhpParamsInspection */
+                $response = $this->processRecurring($single_payment_amount, $feed,
+                    $payPlanPaymentMethod, $planSchedule);
+
+                if (!($response->transactionId > 0 && null !== $response->authorizationCode)) {
+
+                    $this->log_debug(__METHOD__ . '(): Setup Fee Failed!! ');
+
+                    return $this->authorization_error($userError);
+
+                } // if
+
+            } // if
+
+            $subscribResult = array(
+                'is_success' => true,
+                'subscription_id' => $plan->paymentMethodKey,
```

Comment:
Is this supposed to be `$plan->paymentMethodKey`? Seems like `subscription_id` should be set to something specific to the payment schedule.